JSTasks/Tic_Tac_Toe/script.js

```diff
@@ -0,0 +1,187 @@
+const x_image_path = "./images/x_image.png";
+const o_image_path = "./images/o_image.png";
+const background_image = "./images/white_background.png";
+const error_msg_id = "error_msg";
+const player_symbol = "x";
+const bot_symbol = "o";
+const restart_id = "restart_button";
+const max_row = 3;
+const max_col = 3;
+
+const cells_id = [
+    ["row_1_col_1", "row_1_col_2", "row_1_col_3"],
+    ["row_2_col_1", "row_2_col_2", "row_2_col_3"],
+    ["row_3_col_1", "row_3_col_2", "row_3_col_3"]
+];
+
+let cells = [
+    ["", "", ""],
+    ["", "", ""],
+    ["", "", ""]
+];
+
+for (let i = 0; i < max_row; i++) {
+    for (let j = 0; j < max_col; j++) {
+        add_move(i, j);
+    }
+};
+
+document.getElementById(restart_id).addEventListener("click", function() {
+    for (let i = 0; i < max_row; i++) {
+        for (let j = 0; j < max_col; j++) {
+            cells[i][j] = "";
+            document.getElementById(cells_id[i][j]).src = background_image;
+        }
+    }
+
+});
+
+
+function add_move(i, j) {
+    document.getElementById(cells_id[i][j]).addEventListener("click", function() {
+
+        document.getElementById(error_msg_id).innerText = "";
+
+        if (cells[i][j] !== "") {
+            document.getElementById(error_msg_id).innerText = "Please, select valid cell.";
+            return;
+        }
+        cells[i][j] = player_symbol;
+        document.getElementById(cells_id[i][j]).src = x_image_path;
+
+        let winner = check_for_winner();
+        if (winner) {
+            if (winner === player_symbol) {
+                alert("Congratulations! You win.");
```

Comment:
Maybe you can use something else rather than `alert`. Something similar to the error message `<span>` would be a good choice in my opinion.

Reply:
Also, all of the alerts (for winning/losing/end of the game) show up after clicking on a specific cell but before explicitly showing our/bot`s move (the image in the cell), so the cell is still empty. Example:

![alert](https://user-images.githubusercontent.com/19433296/82210622-13e10400-9918-11ea-8b2f-43fe3b19b841.png)